Gems/Terrain/Assets/Passes/TerrainMacroTextureClipmapGenerationPass.pass

```diff
@@ -0,0 +1,287 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "PassAsset",
+    "ClassData": {
+        "PassTemplate": {
+            "Name": "TerrainMacroTextureClipmapGenerationPassTemplate",
+            "PassClass": "TerrainMacroTextureClipmapGenerationPass",
+            "Slots": [
+                {
+                    "Name": "MacroColorClipmapOutput0",
+                    "ShaderInputName": "m_macroColorClipmaps",
+                    "ShaderInputArrayIndex": "0",
```

Comment:
I have changed to Texture2DArray. And I have moved image attachment into C++.
After discussing with Ken, we will get rid of the pyramid part of the clipmaps, for various reasons:
1. We will have clipmap size and stack size configurable, so that the last layer of clipmaps will be enough for far away terrain.
2. Mipmap will make it difficult if we want to scale each clipmap level at power of any other base other than 2.
3. Extra downsample pass adds cost

This means we won't have a texture array with different size or format. So using Texture2DArray is better per your points.

Reply:
Just remember that if yout bind a texture where the array length is 1 that would make it a Texture2d and texture2d sampling will not work if it is bound as a Texture2DArray for metal (it works on dx12/vk). We have the same problem on Decals at the moment. For decals we eventually will switch to texture2d[] texture2d[n] so just make sure we don't have to do this for terrain too or this feature will not work on Macs.